### PR TITLE
[exporter/sumologic]: deprecate options which are going to be removed

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Deprecate options which are going to be removed

**Link to tracking Issue:** #23059

**Testing:** N/A

**Documentation:** update `README.md`

### CODE REVIEW

File: exporter/sumologicexporter/README.md [hunk 1]
```diff
@@ -10,6 +10,61 @@
...
+- `metadata_attributes: [<regex>]`
+- `graphite_template: <template>`
+- `source_category: <template>`
+- `source_name: <template>`
+- `source_host: <template>`
+
+After the new exporter will be moved to this repository:
+
+- `carbon2` and `graphite` are going to be no longer supported and `prometheus` or `otlp` format should be used
+- all resource level attributes are going to be treated as `metadata_attributes`. You can use [Group by Attributes processor](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/groupbyattrsprocessor/) to move attributes from record level to resource level. For example:
```
Comment (sumo-drosiek) [inline exporter/sumologicexporter/README.md:31]:
```suggestion
- all resource level attributes are going to be treated as `metadata_attributes`. You can use [Group by Attributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/groupbyattrsprocessor) to move attributes from record level to resource level. For example:
```

File: exporter/sumologicexporter/README.md [hunk 2]
```diff
@@ -10,6 +10,61 @@
...
+      metadata_attribute:
+        - my_attribute
+  # after switch to new collector
+  processors:
+    groupbyattrs:
+      keys:
+        - my_attribute
+  ```
+
+- Source templates (`source_category`, `source_name` and `source_host`) are going to be removed from the exporter and sources may be set using `_sourceCategory`, `sourceName` or `_sourceHost` resource attributes. We recommend to use [Transform Processor](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/transformprocessor/). For example:
```
Comment (sumo-drosiek) [inline exporter/sumologicexporter/README.md:46]:
```suggestion
- Source templates (`source_category`, `source_name` and `source_host`) are going to be removed from the exporter and sources may be set using `_sourceCategory`, `sourceName` or `_sourceHost` resource attributes. We recommend to use [Transform Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor/). For example:
```

**Review by songy23 — APPROVED** [review]

File: exporter/sumologicexporter/README.md [hunk 3]
```diff
@@ -10,6 +10,61 @@
 [contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
 <!-- end autogenerated section -->
 
+## Migration to new architecture
+
+**This exporter is subject of major changes now.**
```
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/README.md:15]:
```suggestion
**This exporter is undergoing major changes right now.**
```

File: exporter/sumologicexporter/README.md [hunk 4]
```diff
@@ -10,6 +10,61 @@
...
+
+For some time we have been developing [new Sumo Logic exporter](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/exporter/sumologicexporter#sumo-logic-exporter) and now we are in progress of moving it into this repository.
```
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/README.md:17]:
```suggestion
For some time we have been developing the [new Sumo Logic exporter](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/exporter/sumologicexporter#sumo-logic-exporter) and now we are in the process of moving it into this repository.
```

File: exporter/sumologicexporter/README.md [hunk 5]
```diff
@@ -10,6 +10,61 @@
...
+
+The following options are deprecated and they won't exist in new version:
```
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/README.md:19]:
```suggestion
The following options are deprecated and they will not exist in new version:
```

File: exporter/sumologicexporter/README.md [hunk 6]
```diff
@@ -10,6 +10,61 @@
...
+
+- `metric_format: {carbon2, graphite}`
+- `metadata_attributes: [<regex>]`
+- `graphite_template: <template>`
+- `source_category: <template>`
+- `source_name: <template>`
+- `source_host: <template>`
```
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/README.md:26]: We're going to drop some other settings from the distro exporter very soon as well:

- `clear_logs_timestamp`
- `routing_atttribute_to_drop`
- `json_logs.add_timestamp`
- `json_logs.flatten_body`

Do you think we should mention them here as well, even though we haven't dropped them yet in the fork?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/README.md:26]: I wouldn't mention them here, as they never existed in this repository code

**Review by swiatekm — APPROVED** [review]

File: exporter/sumologicexporter/README.md [hunk 7]
```diff
@@ -30,12 +85,18 @@ exporters:
 
     # List of regexes for attributes which should be send as metadata
     # default = []
+    #
+    # This opion is deprecated:
```
Comment (codeboten) [inline exporter/sumologicexporter/README.md:89]:
```suggestion
    # This option is deprecated:
```

File: exporter/sumologicexporter/README.md [hunk 8]
```diff
@@ -44,24 +105,36 @@ exporters:
     #
     # Please regfer to Source temmplates for formatting explanation:
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#source-templates
+    #
+    # This opion is deprecated:
```
Comment (codeboten) [inline exporter/sumologicexporter/README.md:109]:
```suggestion
    # This option is deprecated:
```

File: exporter/sumologicexporter/README.md [hunk 9]
```diff
@@ -44,24 +105,36 @@ exporters:
...
+    # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#migration-to-new-architecture
     graphite_template: <template>
 
     # Desired source category. Useful if you want to override the source category configured for the source.
     #
     # Please regfer to Source temmplates for formatting explanation:
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#source-templates
+    #
+    # This opion is deprecated:
```
Comment (codeboten) [inline exporter/sumologicexporter/README.md:118]:
```suggestion
    # This option is deprecated:
```

File: exporter/sumologicexporter/README.md [hunk 10]
```diff
@@ -44,24 +105,36 @@ exporters:
...
+    # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#migration-to-new-architecture
     source_category: <template>
 
     # Desired source name. Useful if you want to override the source name configured for the source.
     #
     # Please regfer to Source temmplates for formatting explanation:
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#source-templates
+    #
+    # This opion is deprecated:
```
Comment (codeboten) [inline exporter/sumologicexporter/README.md:127]:
```suggestion
    # This option is deprecated:
```

File: exporter/sumologicexporter/README.md [hunk 11]
```diff
@@ -44,24 +105,36 @@ exporters:
...
+    # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#migration-to-new-architecture
     source_name: <template>
 
     # Desired source host. Useful if you want to override the source hosy configured for the source.
     #
     # Please regfer to Source temmplates for formatting explanation:
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter#source-templates
+    #
+    # This opion is deprecated:
```
Comment (codeboten) [inline exporter/sumologicexporter/README.md:136]:
```suggestion
    # This option is deprecated:
```

**Review by codeboten — APPROVED** [review]

**Pull request merged**